python/llm/src/bigdl/llm/transformers/low_bit_linear.py

```diff
@@ -542,15 +542,41 @@ def forward(self, x: torch.Tensor):
             else:
                 from bigdl.llm.utils.isa_checker import is_server, is_spr
 
-                # convert if necessary, and compute a linear result
-                if is_server() and (not is_spr()) and \
-                        self.qtype == SYM_INT4 and x_2d.shape[0] >= TORCH_LINEAR_THRESHOLD:
-                    x0_fp32 = ggml_int4_convert_fp32(x0, self.weight_shape, self.weight_length)
-                    result = F.linear(x, x0_fp32)
+                # # convert if necessary, and compute a linear result
+                # if is_server() and (not is_spr()) and \
+                #         self.qtype == SYM_INT4 and x_2d.shape[0] >= TORCH_LINEAR_THRESHOLD:
+                #     x0_fp32 = ggml_int4_convert_fp32(x0, self.weight_shape, self.weight_length)
+                #     result = F.linear(x, x0_fp32)
+                # else:
+                #     # Weight does not need a convert
+                #     result = ggml_matmul_src1_x_src0_t(x0, x_2d, self.weight_shape, self.qtype)
+                #     result = result.view(new_shape)
+                
+                result = ggml_matmul_src1_x_src0_t(x0, x_2d, self.weight_shape, self.qtype)
+                result = result.view(new_shape)
+                
+                result_0 = ggml_matmul_src1_x_src0_t(x0, x_2d[:x.size(1)], self.weight_shape, self.qtype)
+                new_shape_0 = (1,) + new_shape[1:]
+                result_0 = result_0.view(new_shape_0)
+                
+                if not torch.equal(result_0,result[0].unsqueeze(0)):
```

Comment:
Linear difference is very small.

tensor([[[-5.9605e-08, -8.1956e-08,  5.9605e-08,  ...,  1.6391e-07,
          -7.4506e-09, -2.2352e-08],
         [-5.9605e-08, -8.1956e-08,  5.9605e-08,  ...,  1.6391e-07,
          -7.4506e-09, -2.2352e-08],
         [ 2.2352e-08, -8.9407e-08, -4.7684e-07,  ..., -2.2352e-08,
          -4.1723e-07,  1.3411e-07],
         ...,
         [-2.2352e-08,  0.0000e+00, -1.1921e-07,  ..., -2.9802e-08,
          -2.9802e-08,  7.4506e-08],
         [ 1.4901e-08,  3.5763e-07,  0.0000e+00,  ...,  2.2352e-08,
          -1.4901e-08,  2.9802e-08],
         [ 7.9861e-08,  1.1921e-07, -2.3842e-07,  ...,  1.5646e-07,
           1.1921e-07, -1.4901e-08]]])